pymongo/asynchronous/cursor.py

```diff
@@ -1260,6 +1260,19 @@ async def next(self) -> _DocumentType:
         else:
             raise StopAsyncIteration
 
+    async def _get_all(self, result: list) -> list[_DocumentType]:
+        """Get all documents from the cursor."""
+        if not self._exhaust_checked:
+            self._exhaust_checked = True
+            await self._supports_exhaust()
+        if self._empty:
+            raise StopAsyncIteration
+        while self.alive:
```

Comment:
This loop seems problematic for tailable cursors. I would expect `cursor.to_list()` to has the same behavior as `[x async for x in cursor]`. The former would run forever on a tailable cursor while the later stops when the first empty batch is returned. See https://pymongo.readthedocs.io/en/latest/examples/tailable.html#tailable-cursors for an example.

Reply:
Makes sense, fixed.

Reply:
Could you add a test case for this?